aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalogInteg.java

```diff
@@ -0,0 +1,375 @@
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.ObjectIdentifier;
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+@Ignore
+public class TestGlueCatalogInteg {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TestGlueCatalogInteg.class);
+
+  private static final List<String> NAMESPACES = new ArrayList<>();
+  private static final GlueClient GLUE = GlueClient.create();
+  private static final S3Client S3 = S3Client.create();
+  private static final String PREFIX = getRandomName();
+
+  // the integration test requires the following env variables
+  private static final String AWS_ACCESS_KEY_ID = System.getenv("AWS_ACCESS_KEY_ID");
+  private static final String AWS_SECRET_ACCESS_KEY = System.getenv("AWS_SECRET_ACCESS_KEY");
+  private static final String BUCKET = System.getenv("GLUE_BUCKET");
+
+  private static GlueCatalog glueCatalog;
+  private static GlueCatalog glueCatalogWithSkip;
+
+  @BeforeClass
+  public static void beforeClass() {
+    Configuration configuration = new Configuration();
+    // Iceberg uses Hadoop 2.7.3, and the hadoop-aws module does not support reading environment variable yet
+    configuration.set("fs.s3a.access.key", AWS_ACCESS_KEY_ID);
+    configuration.set("fs.s3a.secret.key", AWS_SECRET_ACCESS_KEY);
+    String testBucketPath = "s3a://" + BUCKET + "/" + PREFIX;
+    glueCatalog = new GlueCatalog(configuration, GLUE, null, testBucketPath, false);
+    glueCatalogWithSkip = new GlueCatalog(configuration, GLUE, null, testBucketPath, true);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    for (String namespace : NAMESPACES) {
+      try {
+        // delete db also delete tables
+        GLUE.deleteDatabase(DeleteDatabaseRequest.builder().name(namespace).build());
+      } catch (Exception e) {
+        // pass
+      }
+    }
+    boolean hasContent = true;
+    while (hasContent) {
+      ListObjectsV2Response response = S3.listObjectsV2(ListObjectsV2Request.builder()
+          .bucket(BUCKET).prefix(PREFIX).build());
+      hasContent = response.hasContents();
+      if (hasContent) {
+        S3.deleteObjects(DeleteObjectsRequest.builder().bucket(BUCKET).delete(Delete.builder().objects(
+            response.contents().stream()
+                .map(obj -> ObjectIdentifier.builder().key(obj.key()).build())
+                .collect(Collectors.toList())
+        ).build()).build());
+      }
+    }
+  }
+
+  @Test
+  public void testNamespace() {
+    // create ns
+    String namespace = getRandomName();
+    NAMESPACES.add(namespace);
+    glueCatalog.createNamespace(Namespace.of(namespace));
+    Database database = GLUE.getDatabase(GetDatabaseRequest.builder().name(namespace).build()).database();
+    Assert.assertEquals(namespace, database.name());
+    // list all
```

Comment:
Can you separate the different calls into test methods? Combining a lot of cases together makes it harder to see what's wrong because failures prevent other test cases from running.